power_grid_model_c/power_grid_model/include/power_grid_model/component/fault.hpp

```diff
@@ -34,6 +35,15 @@ class Fault final : public Base {
           fault_object_{fault_input.fault_object},
           r_f_{is_nan(fault_input.r_f) ? double{} : fault_input.r_f},
           x_f_{is_nan(fault_input.x_f) ? double{} : fault_input.x_f} {
+        if (fault_type_ == FaultType::three_phase) {
+            if (fault_phase_ != FaultPhase::abc)
+                throw InvalidShortCircuitPhases(fault_type_, fault_phase_);
+        }
+        else {
+            if (fault_phase_ == FaultPhase::abc) {
+                throw InvalidShortCircuitPhases(fault_type_, fault_phase_);
+            }
+        }
```

Comment:
@mgovers  please move this check in a separate function. And call the function both in the constructor and the `update` function.